File: result/translate/starchat/java2cpp/c++/CONVERT_DECIMAL_FRACTION_BINARY_NUMBER.cpp
string f_gold ( double num, int k_prec ) {
    string binary = "";
    int Integral = ( int ) num;
    double fractional = num - Integral;
    while ( Integral > 0 ) {
        int rem = Integral % 2;
        binary += to_string( rem ) ;
        Integral /= 2;
    }
    binary = reverse( binary );
    binary += ".";
    while ( k_prec-- > 0 ) {
        fractional *= 2;
        int fract_bit = ( int ) fractional;
        if ( fract_bit == 1 ) {
            fractional -= fract_bit;
            binary += "1";
        }
        else {
            binary += "0";
        }
    }
    return binary;
}

File: dataset/java2cpp/java/CONVERT_DECIMAL_FRACTION_BINARY_NUMBER.java
String f_gold ( double num , int k_prec ) {
    String binary = "" ;
    int Integral = ( int ) num ;
    double fractional = num - Integral ;
    while ( Integral > 0 ) {
        int rem = Integral % 2 ;
        binary += ( ( char ) ( rem + '0' ) ) ;
        Integral /= 2 ;
    }
    binary = new StringBuffer ( binary ) . reverse() . toString() ;
    binary += ( '.' ) ;
    while ( k_prec -- > 0 ) {
        fractional *= 2 ;
        int fract_bit = ( int ) fractional ;
        if ( fract_bit == 1 ) {
        fractional -= fract_bit ;
        binary += ( char ) ( 1 + '0' ) ;
        }
        else {
        binary += ( char ) ( 0 + '0' ) ;
        }
    }
    return binary ;
}